var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -100,7 +100,7 @@ class Gcc(AutotoolsPackage):
     depends_on('isl@0.15:0.18', when='@6:8.9')
     depends_on('isl@0.15:0.20', when='@9:')
     depends_on('zlib', when='@6:')
-    depends_on('libiconv')
+    depends_on('libiconv', type='build')
```

Comment:
To make sure that the build system does not pick it up on non-Darwin:
```suggestion
    depends_on('libiconv', type='build', when='platform=darwin')
```

Reply:
Shouldn’t be build type because it links to the library

Reply:
I had the same issue on my ubuntu19.04 machine trying to build gcc@9.2.0%gcc@9.2.0. 
The patch fixes the issues and I can compile.

Reply:
I also just recompiled our software stack and this PR (plus my suggested change) fixes the problems I was having with GCC.

Reply:
Requested changes to `depends_on` line are in commit 08a06a7.